src/newgrf_railtype.cpp

```diff
@@ -65,6 +65,16 @@
 	return nullptr;
 }
 
+GrfSpecFeature RailTypeResolverObject::GetFeature() const
+{
+	return GSF_RAILTYPES;
+}
+
+uint32 RailTypeResolverObject::GetLocalID() const
+{
+	return this->railtype_scope.rti->label;
```

Comment:
The local id is super weird to get (reverse lookup in railtype_map), and is actually not used anywhere after initialisation. So I agree that "label" is more useful.
However, the method should not be named "GetLocalID" then.